### PR TITLE
CGMES import: fix HVDC active power set point conversion

### PR DESCRIPTION
**Pull request by annetill**
Signed-off-by: Anne Tilloy <anne.tilloy@rte-france.com>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/elements/hvdc/DcLineSegmentConversion.java [hunk 1]
```diff
@@ -101,19 +101,19 @@ private static double getMaxP(double pAC1, double pAC2, HvdcLine.ConvertersMode
         return DEFAULT_MAXP_FACTOR * Math.abs(pAC1);
     }
 
-    private static double getPDc(double pAC1, double pAC2, double poleLossP1, double poleLossP2,
-        HvdcLine.ConvertersMode mode) {
+    private static double getActivePowerSetpoint(double pAC1, double pAC2, double poleLossP1, double poleLossP2,
```
Comment (marqueslanauja) [inline cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/elements/hvdc/DcLineSegmentConversion.java:104]: Maybe add a comment saying that the active power set point of an HVDC should be interpreted on AC side of the rectifier converter

**Review by marqueslanauja — APPROVED** [review]
I reviewed all the code and it is ok. Another way to avoid the duplicated code is to add a new method `getPdc` in the `HvdcLine` interface. The code of the method will be:

```
    public double getPdc() {
        if (getConvertersMode() == ConvertersMode.SIDE_1_RECTIFIER_SIDE_2_INVERTER) {
            return getActivePowerSetpoint() * (1 - converterStation1.getLossFactor() / 100);
        } else {
            return getActivePowerSetpoint() * (1 - converterStation2.getLossFactor() / 100);
        }
    }

```

File: iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/AbstractHvdcConverterStation.java [hunk 2]
```diff
@@ -57,6 +57,15 @@ public T setLossFactor(float lossFactor) {
         return (T) this;
     }
 
+    @Override
+    public HvdcConverterStation getOtherConverterStation() {
+        if (hvdcLine != null) {
+            return hvdcLine.getConverterStation1() == this ? hvdcLine.getConverterStation2() : hvdcLine.getConverterStation1();
+        } else {
+            throw new UnsupportedOperationException();
```
Comment (miovd) [inline iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/AbstractHvdcConverterStation.java:65]: I don't think we should throw an exception, maybe make the method return an optional that can be empty:
```java
public Optional<HvdcConverterStation> getOtherConverterStation() {
        if (hvdcLine != null) {
            return hvdcLine.getConverterStation1() == this ? Optional.of(hvdcLine.getConverterStation2()) : Optional.of(hvdcLine.getConverterStation1());
        } 
     return Optional.empty();
}
```

File: iidm/iidm-mergingview/src/main/java/com/powsybl/iidm/mergingview/AbstractHvdcConverterStationAdapter.java [hunk 3]
```diff
@@ -42,4 +42,14 @@ public HvdcType getHvdcType() {
...
+
+    @Override
+    public HvdcConverterStation getOtherConverterStation() {
+        HvdcLine hvdcLine = getHvdcLine();
+        if (hvdcLine != null) {
+            return hvdcLine.getConverterStation1() == this ? hvdcLine.getConverterStation2() : hvdcLine.getConverterStation1();
+        } else {
+            throw new UnsupportedOperationException();
+        }
+    }
```
Comment (miovd) [inline iidm/iidm-mergingview/src/main/java/com/powsybl/iidm/mergingview/AbstractHvdcConverterStationAdapter.java:54]: Only do: `return getIndex().getHvdcConverterStation(getDelegate().getHvdcConverterStation());`

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/StateVariablesExport.java [hunk 4]
```diff
@@ -441,9 +441,10 @@ private static void writeConverters(Network network, String cimNamespace, XMLStr
         for (HvdcConverterStation<?> converterStation : network.getHvdcConverterStations()) {
             double poleLoss;
             if (CgmesExportUtil.isConverterStationRectifier(converterStation)) {
-                poleLoss = converterStation.getLossFactor() * converterStation.getHvdcLine().getActivePowerSetpoint() / (100 - converterStation.getLossFactor());
```
[the remark this answers is not in the document]
Comment (marqueslanauja) [reply cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/StateVariablesExport.java:444]: We agree. We should use the terminal P to get poleLoss as a first option and only when terminal P is no available use the current code.

[the remark this answers is not in the document]
Comment (annetill) [reply cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/StateVariablesExport.java:444]: Can you please make the fix?

[the remark this answers is not in the document]
Comment (marqueslanauja) [reply cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/StateVariablesExport.java:444]: Yes, I am doing it.

**Pull request merged**